app/models/vm_or_template/operations/configuration.rb

```diff
@@ -73,8 +73,8 @@ def disconnect_floppies
 
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
-    if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+    if options[:datastore] || options[:datastore_id]
```

Comment:
Fixed by https://github.com/ManageIQ/manageiq/pull/15245, closing the current one.